packages/components/src/input-control/reducer/reducer.ts

```diff
@@ -221,7 +221,10 @@ export function useInputControlStateReducer(
 			initialState.value !== currentState.current.value &&
 			! currentState.current.isDirty
 		) {
-			dispatch( { value: initialState.value } );
+			dispatch( {
+				type: actions.CONTROL,
+				payload: { value: initialState.value ?? '' },
```

Comment:
Would it make sense for the `?? ''` part to be moved directly in the reducer? e.g

```js
nextState.value = action.payload.value ?? '';
```

Reply:
It was in the reducer back in ad3cda1. Now reusing `ValuePayload` with `ControlAction` means it would create a type mismatch unless (as done here) the potential `undefined` is eliminated beforehand. The alternatives I can think of:
- add and use a new type that's the same as `ValuePayload` except `value` is optional
- retype `ValuePayload` to make `value` optional

Those are in order of my preference (behind not changing it) but I'm glad to learn if there's a better option and/or of any drawbacks with how it is now.

Reply:
Thank you for the extra context — let's keep it like this. I know that there are implications when fallbacking from `undefined` to `''` (ie. switching controlled/uncontrolled mode), so it's probably better to keep the code as is.

Reply:
One extra thing I noticed — in this `useInputControlStateReducer` there are several `useLayoutEffect` calls

https://github.com/WordPress/gutenberg/blob/cc941d49c9db171a570f19d5afc8034cf522043c/packages/components/src/input-control/reducer/reducer.ts#L202-L229

It would be great if we could add inline comments for each of these `useLayoutEffect` calls explaining what is their purpose / what they are achieving, as the code is quite convoluted and it can be complicated for a developer that is not familiar with `InputControl` to make sense of it — would you be able to do that? Thank you!

Reply:
>…add inline comments for each of these useLayoutEffect calls explaining what is their purpose…

Done in 3b54cff